File: holoeasy-core/src/main/java/org/holoeasy/pool/IHologramPool.java
package org.holoeasy.pool;

import org.holoeasy.HoloEasy;
import org.holoeasy.hologram.Hologram;
import org.jetbrains.annotations.ApiStatus;
import org.jetbrains.annotations.NotNull;

import java.util.Set;

public interface IHologramPool<T extends Hologram> {

    @NotNull HoloEasy getLib();

    @NotNull Set<T> getHolograms();

    @ApiStatus.Experimental
    void destroy();
}

File: holoeasy-core/src/main/kotlin/org/holoeasy/pool/IHologramPool.kt
package org.holoeasy.pool

import org.holoeasy.HoloEasy
import org.holoeasy.hologram.Hologram
import org.jetbrains.annotations.ApiStatus

interface IHologramPool<T : Hologram> {

    val lib : HoloEasy

    val holograms: Set<T>

    @ApiStatus.Experimental
    fun destroy()
}
